Source/items.cpp

```diff
@@ -3219,9 +3219,19 @@ Item *SpawnUnique(_unique_items uid, Point position, bool sendmsg /*= true*/)
 	while (AllItemsList[idx].iItemId != UniqueItems[uid].UIItemId)
 		idx++;
 
-	GetItemAttrs(item, static_cast<_item_indexes>(idx), curlv);
-	GetUniqueItem(*MyPlayer, item, uid);
-	SetupItem(item);
+	if (!gbIsMultiplayer || GenerateRnd(100) < 15) {
+		GetItemAttrs(item, static_cast<_item_indexes>(idx), curlv);
+		GetUniqueItem(*MyPlayer, item, uid);
+		SetupItem(item);
+	} else {
+		if (level)
+			curlv = *level;
+		const ItemData &uniqueItemData = AllItemsList[idx];
+		_item_indexes idx = GetItemIndexForDroppableItem(false, [&uniqueItemData](const ItemData &item) {
+			return item.itype == uniqueItemData.itype;
+		});
+		SetupAllItems(*MyPlayer, item, idx, AdvanceRndSeed(), curlv * 2, 15, true, false, false);

```

Comment:
```suggestion
	if (gbIsMultiplayer) {
		if (level)
			curlv = *level;
		const ItemData &uniqueItemData = AllItemsList[idx];
		_item_indexes idx = GetItemIndexForDroppableItem(false, [&uniqueItemData](const ItemData &item) {
			return item.itype == uniqueItemData.itype;
		});
		SetupAllItems(*MyPlayer, item, idx, AdvanceRndSeed(), curlv * 2, 15, true, false, false);
	}
	if (!gbIsMultiplayer || item._iMagical == ITEM_QUALITY_UNIQUE) {
		GetItemAttrs(item, static_cast<_item_indexes>(idx), curlv);
		GetUniqueItem(*MyPlayer, item, uid);
		SetupItem(item);
```

Reply:
I really like the idea, because we stick with the normal random loot logic instead of creating our own version of it.

I tested, if all uniques could be generated for the giving dungeon level where the quest is located.
But optic amulet don't drop in diablo, because there are no unique amulets in diablo that could be replaced. 😢

Reply:
Should we just let anything drop of amulet ?

Reply:
You mean changing
```cpp
			return item.itype == uniqueItemData.itype;`
```
to
```cpp
			if (IsAnyOf(item.itype, ItemType::Gold, ItemType::Misc))
				return false;
			if (uniqueItemData.itype == ItemType::Amulet)
				return true;
			return item.itype == uniqueItemData.itype;
```
?
That would work (tested it with optic amulet in diablo).

We could also allow this for all uniques. But this would sometimes make little sense (Grisworlds Edge for example).